### DISCUSSION TITLE
Graph layout plugin

### DISCUSSION THREAD

**Discussion by sar5430** [NationalSecurityAgency/ghidra#4238] [General]
Hey, I recently built a plugin that makes ghidra's Function graphs looks kinda like what rizin does. The result is still far from perfect, but I think some of you might be interested in. The repo is here : https://github.com/sar5430/RizinLayout
I hope it'll help as much as it does for me!

**Comment by dragonmacher** [comment]
I intend to take a look at this.  Is the code all original, copied/ported from another project, or even loosely based off of other source code?  I ask because I want to make sure we don't run into any licensing issues.

**Comment by sar5430** [comment]
Regarding vertex placement it has been copied/ported from Rizin. For edges placement it is original. I have to admit I didn't even think about licensing issues when doing this project

**Comment by dragonmacher** [comment]
>  I have to admit I didn't even think about licensing issues when doing this project

No worries. That is not something I would typically consider outside of work-related tasks.

Can you point me to the original code you used (the specific file(s))?

I personally like the nice documentation that Cutter provides about its graph layout.  Cutter's code is GPL, which means we can't pull it into Ghidra.

**Reply by sar5430** [reply]
Sure, most of the code I copied come from the file https://github.com/rizinorg/cutter/blob/dev/src/widgets/GraphGridLayout.cpp

**Comment by dragonmacher** [comment]
We have added a layout that should address the original need mentioned in this discussion.
